app/services/event.rb

```diff
@@ -0,0 +1,36 @@
+##
+# Represents events occurring within the application that we are interested in tracking.
+#
+# At a minimum, an event has a type and a timestamp at which it occurred. It may also have optional
+# metadata that describes what occurred in more detail. Events are asynchronously sent to our data
+# warehouse using a background job. An instance of `Event` can trigger an arbitrary number of
+# events (to allow for potentially expensive computation on initialization of subclasses).
+class Event
+  TABLE_NAME = "events".freeze
+
+  ##
+  # Asynchronously sends an event and its metadata to the data warehouse
+  #
+  # @param [Symbol, String] event_type The type of event (e.g. `:page_view`) to trigger
+  # @param [Hash{Symbol => Object}] data An optional hash of data to include with the event
```

Comment:
Interested to hear how people feel about this level of documentation (using YARD) for "library" style services like this one – too verbose?